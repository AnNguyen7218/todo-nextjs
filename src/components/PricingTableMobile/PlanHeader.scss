@import 'config.gutters';
@import 'config.colors';
@import 'config.media';

.name {
  margin-bottom: gutter(8);
}

.price {
  margin-bottom: gutter(16);
  color: $purple;
}

.description {
  margin-bottom: gutter(16);

  @include break(md) {
    margin-bottom: gutter(12);
  }
}

.button {
  margin-bottom: gutter(24);
}
