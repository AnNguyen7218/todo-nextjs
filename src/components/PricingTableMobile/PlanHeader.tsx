import React from 'react';
import { PlanKey } from 'types/pricingTable';
import { PLANS_HEADER } from 'constants/pricingTable';
import { Button } from 'components';
import css from './PlanHeader.scss';

type Props = {
  planKey: PlanKey;
  isContentSelected: boolean;
};

const PlanHeader: React.FC<Props> = ({ planKey, isContentSelected }) => {
  const { name, price, description } = PLANS_HEADER[planKey];
  return (
    <div>
      <h3 className={css['name']}>{name}</h3>
      <h2 className={css['price']}>{price}</h2>
      <p className={css['description']}>{description}</p>
      <Button
        theme="containedColor"
        color="purple"
        disabled={!isContentSelected}
        className={css['button']}
      >
        Get started
      </Button>
    </div>
  );
};

export default PlanHeader;
