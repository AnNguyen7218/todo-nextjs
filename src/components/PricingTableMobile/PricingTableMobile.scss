@import 'config.gutters';
@import 'config.colors';

.content-container {
  padding-top: gutter(24);
  padding-bottom: gutter(32);
  background-color: $light-gray;
  border-top: 0.5px solid $light-gray-border;
}

.header-container {
  margin-top: gutter(48);
}

.link {
  margin-top: gutter(12);
  margin-bottom: gutter(24);
}
