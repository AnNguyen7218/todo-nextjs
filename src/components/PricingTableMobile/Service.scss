@import 'config.gutters';
@import 'config.colors';
@import 'config.media';

.container {
  display: flex;
  align-items: center;
  margin-bottom: gutter(8);
  width: 100%;

  @include break(md) {
    width: 50%;
  }
}

.icon {
  width: 1.6rem;
  height: 1.6rem;
  color: $green;
  margin-right: gutter(10);
}
