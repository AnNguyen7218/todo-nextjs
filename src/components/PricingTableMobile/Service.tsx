import React from 'react';
import { ServiceRowOrder, PlanService } from 'types/pricingTable';
import Checked from 'assets/icons/ic-checked.svg';
import { ICON } from 'constants/pricingTable';
import css from './Service.scss';

type Props = {
  service: ServiceRowOrder;
  planServices: PlanService;
};

const Service: React.FC<Props> = ({ service, planServices }) => {
  const { serviceKey, name, type } = service;
  const value = planServices[serviceKey];

  if (!value) {
    return null;
  }

  if (type === ICON) {
    return (
      <div className={css['container']}>
        <Checked className={css['icon']} />
        <p>{name}</p>
      </div>
    );
  }

  if (typeof value === 'string' && !isNaN(Number(value))) {
    return (
      <div className={css['container']}>
        <Checked className={css['icon']} />
        <p>
          {value} {name}
        </p>
      </div>
    );
  }
  return (
    <div className={css['container']}>
      <Checked className={css['icon']} />
      <p>{value}</p>
    </div>
  );
};

export default Service;
