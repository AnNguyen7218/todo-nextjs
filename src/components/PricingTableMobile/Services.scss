@import 'config.gutters';
@import 'config.colors';
@import 'config.fonts';
@import 'config.media';

.service-container {
  padding-top: gutter(16);
  padding-bottom: gutter(8);
  border-top: 0.05rem solid $light-gray-border;
}

.header {
  @include body-medium;
  color: $purple;
  margin-bottom: gutter(8);
}

.sub-service-container {
  display: flex;
  flex-wrap: wrap;
  flex-direction: column;

  @include break(md) {
    flex-wrap: wrap;
    flex-direction: row;
    justify-content: space-between;
  }
}
