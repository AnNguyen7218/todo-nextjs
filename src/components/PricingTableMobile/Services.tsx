import React from 'react';
import { PlanKey } from 'types/pricingTable';
import { PLAN_SERVICES, MOBILE_SERVICES_ORDER } from 'constants/pricingTable';
import Service from './Service';
import css from './Services.scss';

type Props = {
  planKey: PlanKey;
};

const Services: React.FC<Props> = ({ planKey }) => {
  const planServices = PLAN_SERVICES[planKey];
  return (
    <div>
      {MOBILE_SERVICES_ORDER.map((service, index) => {
        const { name, subServices } = service;
        return (
          <div key={index} className={css['service-container']}>
            <p className={css['header']}>{name}</p>
            <div className={css['sub-service-container']}>
              {subServices.map((subService, subIndex) => {
                return (
                  <Service
                    service={subService}
                    planServices={planServices}
                    key={`${index}-${subIndex}`}
                  />
                );
              })}
            </div>
          </div>
        );
      })}
    </div>
  );
};

export default Services;
