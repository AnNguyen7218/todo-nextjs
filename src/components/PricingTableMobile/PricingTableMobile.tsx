import React, { useState } from 'react';
import { PLANS_ORDER, PLANS_HEADER } from 'constants/pricingTable';
import { TabSlider, Container, HyperLink } from 'components';
import PlanHeader from './PlanHeader';
import Services from './Services';
import css from './PricingTableMobile.scss';

const INITIAL_TAB = 1;

const tabList: TabSliderType[] = PLANS_ORDER.map(planKey => ({
  icon: PLANS_HEADER[planKey].icon,
}));

const PricingTableMobile = () => {
  const [selectedTab, getSelectedTag] = useState(INITIAL_TAB);

  return (
    <div>
      <Container className={css['header-container']}>
        <h2>Choosing a plan for your business</h2>
        <HyperLink href={''} theme="purple" hasHoverOpacity={false}>
          <p className={css['link']}>
            Enterprise? Contact us for a custom solution
          </p>
        </HyperLink>
      </Container>
      <TabSlider
        tabList={tabList}
        initialTab={INITIAL_TAB}
        sliderContainerClassName={css['content-container']}
        getSelectedTag={getSelectedTag}
      >
        {PLANS_ORDER.map((planKey, index) => {
          const isContentSelected = index === selectedTab;
          return (
            <Container key={index}>
              <PlanHeader
                planKey={planKey}
                isContentSelected={isContentSelected}
              />
              <Services planKey={planKey} />
            </Container>
          );
        })}
      </TabSlider>
    </div>
  );
};

export default PricingTableMobile;
