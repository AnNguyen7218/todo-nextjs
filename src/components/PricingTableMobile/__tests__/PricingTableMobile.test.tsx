import React from 'react';
import { render, cleanup } from '@testing-library/react';
import PricingTableMobile from '../PricingTableMobile';

afterEach(cleanup);

describe('PricingTableMobile Component', () => {
  test('it should match snapshot', () => {
    const { container } = render(<PricingTableMobile />);
    expect(container).toMatchSnapshot();
  });
});
