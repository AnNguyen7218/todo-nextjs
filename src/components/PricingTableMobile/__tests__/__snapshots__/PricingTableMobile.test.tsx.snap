// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PricingTableMobile Component it should match snapshot 1`] = `
<div>
  <div>
    <div
      class="container header-container"
    >
      <h2>
        Choosing a plan for your business
      </h2>
      <a
        class="link link--purple"
        href=""
      >
        <p
          class="link"
        >
          Enterprise? Contact us for a custom solution
        </p>
      </a>
    </div>
    <div
      class=""
    >
      <div
        class="container tab-headers-container"
      >
        <div
          class="header"
          data-testid="tab-0"
        >
          <ic-startup.svg
            classname="icon"
          />
        </div>
        <div
          class="header header--selected"
          data-testid="tab-1"
        >
          <ic-professional.svg
            classname="icon"
          />
        </div>
        <div
          class="header"
          data-testid="tab-2"
        >
          <ic-studio.svg
            classname="icon"
          />
        </div>
        <div
          class="header"
          data-testid="tab-3"
        >
          <ic-rocket.svg
            classname="icon"
          />
        </div>
      </div>
      <div
        class="content-container"
      >
        <div
          class="swiper-container swiper-container-initialized swiper-container-horizontal"
        >
          <div
            class="swiper-wrapper"
            style="webkit-transition-duration: 0ms; transition-duration: 0ms; webkit-transform: translate3d(NaNpx, 0px, 0px); transform: translate3d(NaNpx, 0px, 0px);"
          >
            <div
              class="container swiper-slide"
            >
              <div>
                <h3
                  class="name"
                >
                  Startup
                </h3>
                <h2
                  class="price"
                >
                  Free
                </h2>
                <p
                  class="description"
                >
                  Create a profile and get exposure for your business services
                </p>
                <button
                  class="button button--fixed button--containedColor button--purple button--disabled button"
                  disabled=""
                  type="button"
                >
                  <div
                    class="button-children"
                  >
                    Get started
                  </div>
                </button>
              </div>
              <div>
                <div
                  class="service-container"
                >
                  <p
                    class="header"
                  >
                    Services Showcase
                  </p>
                  <div
                    class="sub-service-container"
                  >
                    <div
                      class="container"
                    >
                      <ic-checked.svg
                        classname="icon"
                      />
                      <p>
                        1
                         
                        Service Listing
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="service-container"
                >
                  <p
                    class="header"
                  >
                    Business Development
                  </p>
                  <div
                    class="sub-service-container"
                  >
                    <div
                      class="container"
                    >
                      <ic-checked.svg
                        classname="icon"
                      />
                      <p>
                        Receive & respond to inquiries
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="service-container"
                >
                  <p
                    class="header"
                  >
                    Customer service
                  </p>
                  <div
                    class="sub-service-container"
                  >
                    <div
                      class="container"
                    >
                      <ic-checked.svg
                        classname="icon"
                      />
                      <p>
                        Email support
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="container swiper-slide"
            >
              <div>
                <h3
                  class="name"
                >
                  Professional
                </h3>
                <h2
                  class="price"
                >
                  $100/year
                </h2>
                <p
                  class="description"
                >
                  Create a profile and get exposure for your business services
                </p>
                <button
                  class="button button--fixed button--containedColor button--purple button"
                  type="button"
                >
                  <div
                    class="button-children"
                  >
                    Get started
                  </div>
                </button>
              </div>
              <div>
                <div
                  class="service-container"
                >
                  <p
                    class="header"
                  >
                    Services Showcase
                  </p>
                  <div
                    class="sub-service-container"
                  >
                    <div
                      class="container"
                    >
                      <ic-checked.svg
                        classname="icon"
                      />
                      <p>
                        3
                         
                        Service Listing
                      </p>
                    </div>
                    <div
                      class="container"
                    >
                      <ic-checked.svg
                        classname="icon"
                      />
                      <p>
                        Verified Profile
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="service-container"
                >
                  <p
                    class="header"
                  >
                    Business Development
                  </p>
                  <div
                    class="sub-service-container"
                  >
                    <div
                      class="container"
                    >
                      <ic-checked.svg
                        classname="icon"
                      />
                      <p>
                        Authentication & verification support
                      </p>
                    </div>
                    <div
                      class="container"
                    >
                      <ic-checked.svg
                        classname="icon"
                      />
                      <p>
                        Verified Client Reviews
                      </p>
                    </div>
                    <div
                      class="container"
                    >
                      <ic-checked.svg
                        classname="icon"
                      />
                      <p>
                        Receive & respond to inquiries
                      </p>
                    </div>
                    <div
                      class="container"
                    >
                      <ic-checked.svg
                        classname="icon"
                      />
                      <p>
                        1
                         
                        Sub accounts
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="service-container"
                >
                  <p
                    class="header"
                  >
                    Customer service
                  </p>
                  <div
                    class="sub-service-container"
                  >
                    <div
                      class="container"
                    >
                      <ic-checked.svg
                        classname="icon"
                      />
                      <p>
                        Email support
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="container swiper-slide"
            >
              <div>
                <h3
                  class="name"
                >
                  Studio
                </h3>
                <h2
                  class="price"
                >
                  $400/year
                </h2>
                <p
                  class="description"
                >
                  Create a profile and get exposure for your business services
                </p>
                <button
                  class="button button--fixed button--containedColor button--purple button--disabled button"
                  disabled=""
                  type="button"
                >
                  <div
                    class="button-children"
                  >
                    Get started
                  </div>
                </button>
              </div>
              <div>
                <div
                  class="service-container"
                >
                  <p
                    class="header"
                  >
                    Services Showcase
                  </p>
                  <div
                    class="sub-service-container"
                  >
                    <div
                      class="container"
                    >
                      <ic-checked.svg
                        classname="icon"
                      />
                      <p>
                        10
                         
                        Service Listing
                      </p>
                    </div>
                    <div
                      class="container"
                    >
                      <ic-checked.svg
                        classname="icon"
                      />
                      <p>
                        Verified Profile
                      </p>
                    </div>
                    <div
                      class="container"
                    >
                      <ic-checked.svg
                        classname="icon"
                      />
                      <p>
                        Featured Profile & Services
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="service-container"
                >
                  <p
                    class="header"
                  >
                    Business Development
                  </p>
                  <div
                    class="sub-service-container"
                  >
                    <div
                      class="container"
                    >
                      <ic-checked.svg
                        classname="icon"
                      />
                      <p>
                        Authentication & verification support
                      </p>
                    </div>
                    <div
                      class="container"
                    >
                      <ic-checked.svg
                        classname="icon"
                      />
                      <p>
                        Verified Client Reviews
                      </p>
                    </div>
                    <div
                      class="container"
                    >
                      <ic-checked.svg
                        classname="icon"
                      />
                      <p>
                        Receive & respond to inquiries
                      </p>
                    </div>
                    <div
                      class="container"
                    >
                      <ic-checked.svg
                        classname="icon"
                      />
                      <p>
                        3
                         
                        Sub accounts
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="service-container"
                >
                  <p
                    class="header"
                  >
                    Customer service
                  </p>
                  <div
                    class="sub-service-container"
                  >
                    <div
                      class="container"
                    >
                      <ic-checked.svg
                        classname="icon"
                      />
                      <p>
                        Chat support
                      </p>
                    </div>
                    <div
                      class="container"
                    >
                      <ic-checked.svg
                        classname="icon"
                      />
                      <p>
                        Legal Advisory service
                      </p>
                    </div>
                    <div
                      class="container"
                    >
                      <ic-checked.svg
                        classname="icon"
                      />
                      <p>
                        Product/Profile Advisory service
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="container swiper-slide"
            >
              <div>
                <h3
                  class="name"
                >
                  Growth
                </h3>
                <h2
                  class="price"
                >
                  $900/year
                </h2>
                <p
                  class="description"
                >
                  Create a profile and get exposure for your business services
                </p>
                <button
                  class="button button--fixed button--containedColor button--purple button--disabled button"
                  disabled=""
                  type="button"
                >
                  <div
                    class="button-children"
                  >
                    Get started
                  </div>
                </button>
              </div>
              <div>
                <div
                  class="service-container"
                >
                  <p
                    class="header"
                  >
                    Services Showcase
                  </p>
                  <div
                    class="sub-service-container"
                  >
                    <div
                      class="container"
                    >
                      <ic-checked.svg
                        classname="icon"
                      />
                      <p>
                        10
                         
                        Service Listing
                      </p>
                    </div>
                    <div
                      class="container"
                    >
                      <ic-checked.svg
                        classname="icon"
                      />
                      <p>
                        Verified Profile
                      </p>
                    </div>
                    <div
                      class="container"
                    >
                      <ic-checked.svg
                        classname="icon"
                      />
                      <p>
                        Featured Profile & Services
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="service-container"
                >
                  <p
                    class="header"
                  >
                    Business Development
                  </p>
                  <div
                    class="sub-service-container"
                  >
                    <div
                      class="container"
                    >
                      <ic-checked.svg
                        classname="icon"
                      />
                      <p>
                        Authentication & verification support
                      </p>
                    </div>
                    <div
                      class="container"
                    >
                      <ic-checked.svg
                        classname="icon"
                      />
                      <p>
                        Verified Client Reviews
                      </p>
                    </div>
                    <div
                      class="container"
                    >
                      <ic-checked.svg
                        classname="icon"
                      />
                      <p>
                        Client review with follow up
                      </p>
                    </div>
                    <div
                      class="container"
                    >
                      <ic-checked.svg
                        classname="icon"
                      />
                      <p>
                        Receive & respond to inquiries
                      </p>
                    </div>
                    <div
                      class="container"
                    >
                      <ic-checked.svg
                        classname="icon"
                      />
                      <p>
                        10
                         
                        Sub accounts
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="service-container"
                >
                  <p
                    class="header"
                  >
                    Customer service
                  </p>
                  <div
                    class="sub-service-container"
                  >
                    <div
                      class="container"
                    >
                      <ic-checked.svg
                        classname="icon"
                      />
                      <p>
                        Dedicated account manager
 + Chat support
                      </p>
                    </div>
                    <div
                      class="container"
                    >
                      <ic-checked.svg
                        classname="icon"
                      />
                      <p>
                        Legal Advisory service
                      </p>
                    </div>
                    <div
                      class="container"
                    >
                      <ic-checked.svg
                        classname="icon"
                      />
                      <p>
                        Product/Profile Advisory service
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <span
            aria-atomic="true"
            aria-live="assertive"
            class="swiper-notification"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
